lib/src/ruby/module.cc

```diff
@@ -475,15 +477,21 @@ namespace facter { namespace ruby {
         if (_collection.empty()) {
             bool include_ruby_facts = true;
             _collection.add_default_facts(include_ruby_facts);
-            _collection.add_external_facts(_external_search_paths);
-
+            if (_load_external && !_external_facts_loaded) {
+                _collection.add_external_facts(_external_search_paths);
+                _external_facts_loaded = true;
+            }
```

Comment:
is this needed, as it will always get evaluated in :491?

Reply:
Yep, it was not needed.